### PR TITLE
Run GPU e2e tests on GKE.

### PR DESCRIPTION
**Pull request by mindprince**
Fixes kubernetes/kubernetes#47388.

### CODE REVIEW

**Comment by mindprince** [comment]
This is WIP. This will go green once we release the new gcloud version which supports `--accelerator`.

cc - @vishh

File: jobs/ci-kubernetes-e2e-gke-gpu.env [hunk 1]
```diff
@@ -0,0 +1,10 @@
+### job-env
+PROJECT=k8s-jkns-e2e-gke-gpus
+GKE_CREATE_FLAGS=--accelerator=type=nvidia-tesla-k80,count=2
+KUBE_GKE_ENABLE_KUBERNETES_ALPHA=true
+KUBE_GKE_IMAGE_TYPE=gci
+GINKGO_TEST_ARGS=--ginkgo.focus=\[Feature:GPU\]
+GCLOUD=gcloud
+CMD_GROUP=alpha --quiet
+# GPUs are not available across all zones yet.
```
Comment (vishh) [inline jobs/ci-kubernetes-e2e-gke-gpu.env]: and "it needs additional quota by default"

Comment (mindprince) [inline jobs/ci-kubernetes-e2e-gke-gpu.env]: Updated.

**Comment by mindprince** [comment]
Can you okay to test this?

**Comment by krzyzacy** [comment]
@k8s-bot ok to test

File: jobs/config.json [hunk 2]
```diff
@@ -4702,6 +4702,19 @@
       "UNKNOWN"
     ]
   }, 
+  "ci-kubernetes-e2e-gke-gpu": {
+    "args": [
+      "--env-file=jobs/platform/gke.env",
+      "--env-file=jobs/ci-kubernetes-e2e-gke-gpu.env",
+      "--mode=local",
+      "--timeout=180m",
+      "--extract=ci/latest"
```
Comment (krzyzacy) [inline jobs/config.json]: add `--check-leaked-resources=true` here, and also run `jobs/config_sort.py`

**Pull request merged**